src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,511 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+        if (Object.getOwnPropertyNames(validation).length > 0) {
+            setAddDiskInProgress(false);
+            return;
         }
 
-        return (
-            <Modal position="top" variant="medium" id={`${idPrefix}-dialog-modal-window`} isOpen onClose={Dialogs.close}
-                   title={isMediaInsertion ? _("Insert disc media") : _("Add disk")}
-                   footer={
-                       <>
-                           <Button id={`${idPrefix}-dialog-add`}
-                                   variant='primary'
-                                   isLoading={this.state.addDiskInProgress || this.state.verificationInProgress}
-                                   isDisabled={this.state.addDiskInProgress || this.state.verificationInProgress || dialogLoading ||
-                                               (storagePools.length == 0 && this.state.mode != CUSTOM_PATH) ||
-                                               (this.state.mode == CUSTOM_PATH && this.state.device === "disk" && this.state.customDiskVerificationFailed)}
-                                   onClick={isMediaInsertion ? this.onInsertClicked : this.onAddClicked}>
-                               {isMediaInsertion ? _("Insert") : _("Add")}
-                           </Button>
-                           <Button id={`${idPrefix}-dialog-cancel`} variant='link' onClick={Dialogs.close}>
-                               {_("Cancel")}
-                           </Button>
-                       </>
-                   }>
-                {this.state.dialogError && <ModalError dialogError={this.state.dialogError} dialogErrorDetail={this.state.dialogErrorDetail} />}
-                {defaultBody}
-            </Modal>
-        );
-    }
-}
+        setAddDiskInProgress(true);
+
+        return isMediaInsertion
+            ? onInsertClicked()
+            : onAddClicked()
+                    .then(() => { // force reload of VM data, events are not reliable (i.e. for a down VM)
+                        Dialogs.close();
+                        return domainGet({ connectionName: vm.connectionName, name: vm.name, id: vm.id });
+                    })
+                    .catch(exc => {
+                        setAddDiskInProgress(false);
+                        dialogErrorSet(_("Disk failed to be added"), exc.message);
+                    });
+    };
+
+    return (
+        <>
+            <Button id={`${idPrefix}-dialog-add`}
+                    variant='primary'
+                    isLoading={addDiskInProgress || verificationInProgress}
+                    isDisabled={addDiskInProgress || verificationInProgress || dialogLoading ||
+                                (storagePools.length == 0 && mode != CUSTOM_PATH) ||
+                                (mode == CUSTOM_PATH && diskParams.device === "disk" && customDiskVerificationFailed)}
+                    onClick={onClick}>
+                {isMediaInsertion ? _("Insert") : _("Add")}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231110-154059-d19bea72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)